### ISSUE TITLE
What changes were necessary for EmojiArt for Xcode 9.3?

### ISSUE THREAD

**Issue by hugofalkman** [BestKora/CS193P-Fall-2017-DEMO#1]
EmojiArt L12 in the version from Stanford  no longer in Xcode 9.3 can vary the size of the Emojis -- they are dropped on the background in a constant (small) size independent of the level of zooming for the background.

That fixed size also stays the same when trying to use the pinch gesture.

i will analyze what you did to make your version  of EmojiArt L12 work correctly in Xcode 9.3 but any quick hints what you changed would be much appreciated.

**Comment by hugofalkman** [comment]
Ok, see it's a bug in Xcode 9.3 where the Interaction Drop Delegate loses the NSAttributedString font information passed by the CollectionView Drag Delegate.

Your fix is clearly commented in the EmojiArtView file.

**Comment by BestKora** [comment]
This error only appears in the collection. 
I added a minimum amount of code in  EmojiArtView.swift, so that if the error was corrected, it would be easy to remove it
<img width="776" alt="screen shot 2018-07-14 at 12 26 29" src="https://user-images.githubusercontent.com/8070711/42723612-7720176a-8761-11e8-92ec-86580d286600.png">

I did not see anyone reporting this error, which concerns only the Collection View and NSAttributedString.

**Comment by hugofalkman** [comment]
Thank you. Did you notice that when you run EmojiArt in the 11.0.1 and 11.4 simulators, the dropped emojis are displayed significantly larger by 11.4? Am not sure it's a bug but it would be interesting to know why they are larger in 11.4.

**Comment by BestKora** [comment]
For me this depends on the size or resolution of the background receiving image:
<img width="769" alt="screen shot 2018-07-16 at 15 47 33" src="https://user-images.githubusercontent.com/8070711/42765071-7a816e10-8917-11e8-83c4-0a17730ad7b1.png">
<img width="776" alt="screen shot 2018-07-16 at 15 51 07" src="https://user-images.githubusercontent.com/8070711/42765211-c88e75c6-8917-11e8-8296-a07e5baf1a12.png">

The same situation on device.

**Comment by hugofalkman** [comment]
Yes, agree it depends on the size and resolution of the background image. What I found intriguing was that with the same background image and the same simulator device going from ios 11.0.1 to 11.4 increases the size of the dropped emojis.These screenshots are of your code on the same simulator device with ios 11.0.1 and 11.4 respectively.

![simulator screen shot - ipad pro 12 9-inch - 2018-07-16 at 17 32 55](https://user-images.githubusercontent.com/10969068/42768097-ec472d6c-891e-11e8-9801-f381f2d1f163.png)

![simulator screen shot - ipad pro 12 9-inch - 2018-07-16 at 17 35 04](https://user-images.githubusercontent.com/10969068/42768139-01bdd628-891f-11e8-9d53-6373ffd2f509.png)

**Comment by BestKora** [comment]
iOS 11.0 - version, with which Professor worked and I too at the begining of study Lecture 12, then I went to Lectire 13, it was already iOS 11.2 - and everything has broken. I found, that NSAttributedString has lost font (nil), I have to do something to continue my study.
Try remove my code in iOS 11.0 - it must work naturally without any support for size of font. I don't have this opportunity.
On simulator 11.3 the same situation  as on iOS 11.4 - change background image by Draging new one - change size of emoji:
<img width="763" alt="screen shot 2018-07-16 at 17 30 31" src="https://user-images.githubusercontent.com/8070711/42768678-76d8ed3e-8920-11e8-8c01-1b7f02889d0d.png">
<img width="780" alt="screen shot 2018-07-16 at 17 31 40" src="https://user-images.githubusercontent.com/8070711/42768703-8364e83c-8920-11e8-9107-0d27f6b5dbe6.png">
<img width="759" alt="screen shot 2018-07-16 at 17 32 27" src="https://user-images.githubusercontent.com/8070711/42768723-905a494c-8920-11e8-870e-c2d476d8926e.png">

**Comment by hugofalkman** [comment]
Your code works fine in ios 11.0.1. You cleverly designed your fix so that the code will work once the bug is corrected -- that makes it work correctly also in 11.0.1.

You can just add the 11.0 simulator in Xcode preferences, components tab. 

Edit: I did test your code in 11.0.1 with just

    label.attributedText = NSAttributedString(string: attributedString.string,attributes: [.font:self.font])

in the addLabel method, and it generated the same smaller emojis as the Stanford downloaded code. So your code is not the reason for the larger emojis in 11.4, it is something with the system (or possibly with the different Safari releases generating different resolutions in its drag/drop support.)

**Comment by BestKora** [comment]
Thanks. For study purposes it's OK. But we must report Apple about this bug for the current version of iOS 11?

**Comment by hugofalkman** [comment]
Ok, I continued testing different simulator devices and I find the larger emojis on ios 11.4 only happen on the iPad Pro 12.9 first generation simulator. My screenshots happened to be from from this device.

But on all other iPad simulators I tested the emojis were the same size as for ios 11.0.1. For instance iPad Pro 12.9 second generation, iPad Pro 10.5 and 9.7, as well as iPad 5th generation and iPad Air 2.

So I guess it's a bug in that specific simulator, and not really worth pursuing further.
